functions/go/create-setters/createsetters/create_setters.go

```diff
@@ -0,0 +1,275 @@
+
+// CreateSetters creates a comment for the resource fields which 
+// contains the same value as setter reference value 
+type CreateSetters struct {
+	// Setters holds the user provided values for simple map setters
+	Setters []Setter
+
+	// Setters holds the user provided values for array setters
+	ArraySetters []ArraySetter
+
+	// Results are the results of applying setter values
+	Results []*Result
+
+	// filePath file path of resource
+	filePath string
+}
+
+// Setter stores name and value of the map setter
+type Setter struct {
+	// Name is the name of the setter
+	Name string
+
+	// Value is the input value for setter
+	Value string
+}
+
+// ArraySetter stores name and values of the array setter
+type ArraySetter struct {
+	// Name is the name of the setter
+	Name string
+
+	// ValueSet is the set of the values for setter
+	ValueSet map[string]bool
+}
+
+// Result holds result of search and replace operation
+type Result struct {
+	// FilePath is the file path of the matching value
+	FilePath string
+
+	// FieldPath is field path of the matching value
+	FieldPath string
+
+	// Comment of the matching value
+	Value string
+}
+
+// Filter implements Set cs a yaml.Filter
+func (cs *CreateSetters) Filter(nodes []*yaml.RNode) ([]*yaml.RNode, error) {
+	if len(cs.Setters) == 0 {
+		return nodes, fmt.Errorf("input setters list cannot be empty")
+	}
+	for i := range nodes {
+		filePath, _, err := kioutil.GetFileAnnotations(nodes[i])
+		if err != nil {
+			return nodes, err
+		}
+		cs.filePath = filePath
+		err = accept(cs, nodes[i])
+		if err != nil {
+			return nil, errors.Wrap(err)
+		}
+	}
+	return nodes, nil
+}
+
+/**
+visitMapping takes the mapping node and performs following steps,
+checks if it is a sequence node 
+checks if all the values in the node are present to any of the ArraySetters
+adds the linecomment if they are equal 
+
+e.g. for input of Mapping node
+
+environments: 
+  - dev
+  - stage
+
+For input ApplySetters [Name: env, ValueSet: [dev, stage]], yaml node is transformed to
+
+environments: # kpt-set: ${env}
+  - dev
+  - stage
+
+e.g. for input of Mapping node with FlowStyle
+
+env: [foo, bar]
+
+For input ApplySetters [Name: env, ValueSet: [foo, bar]], yaml node is transformed to
+
+env: [foo, bar] # kpt-set: ${env}
+*/
+
+func (cs *CreateSetters) visitMapping(object *yaml.RNode, path string) error {
+	return object.VisitFields(func(node *yaml.MapNode) error {
+		if node == nil || node.Key.IsNil() || node.Value.IsNil() {
+			// don't do IsNilOrEmpty check cs empty sequences are allowed
+			return nil
+		}
+		
+		// the aim of this method is to create-setter for sequence nodes
+		if node.Value.YNode().Kind != yaml.SequenceNode {
+			// return if it is not a sequence node
+			return nil
+		}
+
+		// checks if the kind is flowstyle and adds comment to its value node 
+		// else it adds the comment to the key node 
+		changeNodeComment := node.Key
+
+		if node.Value.YNode().Style == yaml.FlowStyle {
```

Comment:
Unless the sequence node is empty, standardize sequence node to FoldedStyle and then add comment to key node. Check  https://github.com/GoogleContainerTools/kpt-functions-catalog/blob/master/functions/go/apply-setters/applysetters/apply_setters.go#L161